### ISSUE TITLE
Mouse drag broken using SweetHome3D

### ISSUE THREAD

**Issue by avrelaun** [swaywm/sway#3664]
context:
- archlinux
- sway version 1.0-rc1-45-gfb4f2928 (Feb 12 2019, branch 'master')
- xwayland
- sweethome3d from arch repos

I have been using sway since the first 1.0 beta and building / running it every week. I began experiencing an issue with SweetHome3D that I did not have with i3 or with the first sway 1.0 betas IIRC. To reproduce:

- Launch SweetHome3D
- Click on the wall creation button on the toolbar
- Try creating a wall by dragging the mouse on the drawing area
-> The wall begins appearing under the mouse but wuickly disappears after a a second even while holding the mouse button. It seems to be time-related and not distance-related.

**Comment by avrelaun** [comment]
This is really annoying, and it's pretty much the only reason I still have to start i3.
I suspect all desktop Java apps might have a similar behaviour.

**Comment by cycomanic** [comment]
I have just encountered this bug as well. I think it is due to SweetHome3D displaying a "tooltip/overlay" (not sure what the correct term is) with information about the measurements. It works fine under gnome with wayland, so doesn't seem to be a wayland issue.

**Comment by m-khvoinitsky** [comment]
Yes, this is happening because tooltips steal focus from parent window which leads to stopping of drag move.
This almost fixes the issue:
```
no_focus [class="com-eteks-sweethome3d-SweetHome3D" instance="sun-awt-X11-XWindowPeer"]
```
However, the problem still reproduces if you hit the tooltip with a mouse, unfortunately, I don't know how to make window never receive focus.

**Comment by emersion** [comment]
An `xprop` dump of the tooltip window would be helpful.

**Comment by m-khvoinitsky** [comment]
That was tricky, but I've managed to do:
```
WM_STATE(WM_STATE):
		window state: Withdrawn
		icon window: 0x0
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_DIALOG
_NET_WM_STATE(ATOM) = 
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: False
		window id # of group leader: 0x9e00074
WM_TRANSIENT_FOR(WINDOW): window id # 0x9e00074
_NET_WM_PID(CARDINAL) = 28942
WM_CLIENT_MACHINE(STRING) = "m-khvoinitsky-arch-mbp"
WM_PROTOCOLS(ATOM): protocols  
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x3, 0x0, 0x0, 0x0, 0x0
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 1279, 742
		program specified size: 205 by 95
		window gravity: NorthWest
WM_CLASS(STRING) = "sun-awt-X11-XWindowPeer", "com-eteks-sweethome3d-SweetHome3D"
WM_CLIENT_LEADER(WINDOW): window id # 0x9e00008
_NET_WM_NAME(UTF8_STRING) = "win3"
WM_NAME(STRING) = "win3"
```

**Comment by emersion** [comment]
I wonder why this isn't set as an override-redirect window.

**Comment by spectrum70** [comment]
any news on this ? I have same issue on 7.0.2, Xwayland and wayfire compositor

**Comment by janhieber** [comment]
For Hyprland use: `windowrulev2 = nofocus,class:(com-eteks-sweethome3d-SweetHome3D)`

**Comment by ghostface** [comment]
FYI: I have no such problems with sway 1.9-4 and sweethome3d 7.4-1

**Comment by DavidHenryThoreau** [comment]
@ghostface Are you using `xorg-xwayland` ?

**Comment by ghostface** [comment]
> @ghostface Are you using `xorg-xwayland` ?

yes - xorg-xwayland 24.1.4-1

**Comment by DavidHenryThoreau** [comment]
Right, I've tested too, it's working for me, thanks for feedback.

**Comment by m-breitbach** [comment]
https://github.com/NaharEmet/sweethome3d-7.5-wayland-patch
